pkg/controller/atlasproject/cloud_provider_access.go

```diff
@@ -0,0 +1,212 @@
+package atlasproject
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.uber.org/zap"
+
+	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+func ensureProviderAccessStatus(context *workflow.Context, project *v1.AtlasProject, groupID string) workflow.Result {
+	roleStatuses := project.Status.DeepCopy().CloudProviderAccessRoles
+	roleSpecs := project.Spec.DeepCopy().CloudProviderAccessRoles
+
+	result, condition := syncProviderAccessStatus(context, roleSpecs, roleStatuses, groupID)
+	if result != workflow.OK() {
+		context.SetConditionFromResult(condition, result)
+		return result
+	}
+	context.SetConditionTrue(status.CloudProviderAccessReadyType)
+	return result
+}
+
+func syncProviderAccessStatus(customContext *workflow.Context, specs []v1.CloudProviderAccessRole, statuses []status.CloudProviderAccessRole, groupID string) (workflow.Result, status.ConditionType) {
+	client := customContext.Client
+	logger := customContext.Log
+	ctx := context.Background() // TODO: eoaueoaueoa
```

Comment:
This line seems unfinished at the end part 😂

Reply:
Fixed